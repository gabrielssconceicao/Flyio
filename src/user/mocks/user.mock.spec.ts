import {
  generateCreateUserDtoMock,
  generateFindAllUsersResponseDtoMock,
  generateUserMock,
} from './user.mock';

describe('<User Mocks/ >', () => {
  it('should generate a valid CreateUserDto mock without bio', () => {
    const dto = generateCreateUserDtoMock();

    expect(dto).toEqual({
      name: 'John Doe',
      username: 'jDoe453',
      email: 'jdoe@me.com',
      password: '123456',
    });
  });

  it('should generate a valid CreateUserDto mock with bio', () => {
    const dto = generateCreateUserDtoMock(true);

    expect(dto).toEqual({
      name: 'John Doe',
      username: 'jDoe453',
      email: 'jdoe@me.com',
      password: '123456',
      bio: 'This is my bio',
    });
  });

  it('should generate a valid FindAllUsersResponseDto mock', () => {
    const responseDto = generateFindAllUsersResponseDtoMock();

    expect(responseDto).toEqual({
      count: 1,
      users: [
        {
          id: '42-d-f-df4',
          name: 'John Doe',
          username: 'jDoe453',
          profileImg: 'https://example.com/profile.jpg',
        },
      ],
    });
  });

  it('should generate a valid User mock', () => {
    const user = generateUserMock();

    expect(user).toEqual({
      id: '42-d-f-df4',
      name: 'John Doe',
      username: 'jDoe453',
      email: 'jdoe@me.com',
      profileImg: 'https://example.com/profile.jpg',
      bio: 'This is my bio',
      active: true,
    });
  });
});
