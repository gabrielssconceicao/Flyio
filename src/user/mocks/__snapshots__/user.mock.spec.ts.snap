// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMocks /> should generate a valid user mock 1`] = `
{
  "active": true,
  "bio": "This is my bio",
  "email": "jdoe@me.com",
  "id": "42-d-f-df4",
  "name": "John Doe",
  "profileImg": "https://example.com/profile.jpg",
  "username": "jDoe453",
}
`;
